extensions/refresh-catalog/src/test/java/org/eclipse/dataspaceconnector/mvd/FederatedCacheNodeResolverTest.java

```diff
@@ -0,0 +1,91 @@
+import org.eclipse.dataspaceconnector.iam.did.spi.document.DidDocument;
+import org.eclipse.dataspaceconnector.iam.did.spi.document.Service;
+import org.eclipse.dataspaceconnector.iam.did.spi.resolution.DidResolverRegistry;
+import org.eclipse.dataspaceconnector.registration.client.models.Participant;
+import org.eclipse.dataspaceconnector.spi.EdcException;
+import org.eclipse.dataspaceconnector.spi.monitor.ConsoleMonitor;
+import org.eclipse.dataspaceconnector.spi.result.Result;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class FederatedCacheNodeResolverTest {
+
+    FederatedCacheNodeResolver resolver;
+    DidResolverRegistry didResolver = mock(DidResolverRegistry.class);
+    static final Faker FAKER =  Faker.instance();
+    String did = "did:web:" + FAKER.internet().domainName();
+    static String idsUrl = FAKER.internet().url();
+
+    @ParameterizedTest
+    @MethodSource("propertiesSource")
+    void getNode_success(List<Service> services) {
+        when(didResolver.resolve(did)).thenReturn(Result.success(getDidDocument(services)));
+
+        resolver =  new FederatedCacheNodeResolver(didResolver, new ConsoleMonitor());
+
+        var node = resolver.toFederatedCacheNode(new Participant().did(did));
+
+        assertThat(node.getName()).isEqualTo(did);
+        assertThat(node.getTargetUrl()).isEqualTo(idsUrl);
+        assertThat(node.getSupportedProtocols()).containsExactly("ids-multipart");
+    }
+
+    @Test
+    void getNode_failure() {
+        when(didResolver.resolve(did)).thenReturn(Result.failure("failure"));
+
+        resolver =  new FederatedCacheNodeResolver(didResolver, new ConsoleMonitor());
+
+        assertThatThrownBy(() -> resolver.toFederatedCacheNode(new Participant().did(did))).isInstanceOf(EdcException.class);
+    }
+
+    @Test
+    void getNode_noIdsUrl() {
+        when(didResolver.resolve(did)).thenReturn(Result.success(getDidDocument(List.of())));
+
+        resolver =  new FederatedCacheNodeResolver(didResolver, new ConsoleMonitor());
+
+        assertThatThrownBy(() -> resolver.toFederatedCacheNode(new Participant().did(did))).isInstanceOf(EdcException.class);
+    }
+
+    @NotNull
+    private DidDocument getDidDocument(List<Service> services) {
+        return DidDocument.Builder.newInstance().id(did).service(services).build();
+    }
+
+    private static Stream<Arguments> propertiesSource() {
+        return Stream.of(
+                Arguments.of(List.of(new Service("#idsUrl", FederatedCacheNodeResolver.IDS_URL, idsUrl))),
```

Comment:
```suggestion
                arguments(List.of(new Service("#idsUrl", FederatedCacheNodeResolver.IDS_URL, idsUrl))),
```
nit: with static import - I find this form more readable